[BUG] Not able to record events in "unit_testing" approach

<!--- Provide a general summary of the issue in the Title above. -->

## Protostar version
<!--- Tell us what Protostar version you are running. -->
<!--- You can check the Protostar and Cairo-lang version by running `protostar -v`. -->
| Name       | Version              |
| ---------- | -------------------- |
| Protostar  | `PROTOSTAR_VERSION`  |
| Cairo-lang | `CAIRO_LANG_VERSION` |
| Git        | `GIT_VERSION`        |


## Current Behavior
self = <protostar.commands.test.cheatcodes.expect_events_cheatcode.ExpectEventsCheatcode object at 0x110f1d848>
raw_expected_event = {'data': {'amount': 20, 'current_balance': 123}, 'name': 'increase_balance_called'}

    def _convert_raw_expected_event_to_expected_event(
        self,
        raw_expected_event: RawExpectedEventType,
    ):
    
        name: str
        data: Optional[
            Union[
                List[int],
                Dict[
                    DataTransformerFacade.ArgumentName,
                    DataTransformerFacade.SupportedType,
                ],
            ]
        ] = None
        from_address: Optional[int] = None
        if isinstance(raw_expected_event, str):
            name = raw_expected_event
        else:
            name = raw_expected_event["name"]
            if "data" in raw_expected_event:
                raw_data = raw_expected_event["data"]
                if isinstance(raw_data, collections.Mapping):
                    assert (
                        name in self.state.event_name_to_contract_abi_map
>                   ), "Couldn't map event name to the contract path with that event"
E                   AssertionError: Couldn't map event name to the contract path with that event



## Expected Behavior
It should not crash


## Steps to Reproduce


## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug. -->

--- a/protostar/commands/test/cheatcodes/declare_cheatcode.py
+++ b/protostar/commands/test/cheatcodes/declare_cheatcode.py
@@ -6,7 +6,7 @@ from typing import Any, Callable
 from starkware.python.utils import from_bytes
 from starkware.starknet.business_logic.internal_transaction import InternalDeclare
 from starkware.starknet.testing.contract import DeclaredClass
-from starkware.starknet.testing.contract_utils import EventManager, get_abi
+from starkware.starknet.testing.contract_utils import get_abi
 
 from protostar.starknet.cheatcode import Cheatcode
 from protostar.utils.starknet_compilation import StarknetCompiler
@@ -56,15 +56,7 @@ class DeclareCheatcode(Cheatcode):
             )
 
         abi = get_abi(contract_class=contract_class)
-        event_manager = EventManager(abi=abi)
-        self.state.update_event_selector_to_name_map(
-            # pylint: disable=protected-access
-            event_manager._selector_to_name
-        )
-        # pylint: disable=protected-access
-        for event_name in event_manager._selector_to_name.values():
-            self.state.event_name_to_contract_abi_map[event_name] = abi
-
+        self._add_event_abi_to_state(abi)
         class_hash = tx_execution_info.call_info.class_hash
         assert class_hash is not None
         return DeclaredClass(

--- a/protostar/starknet/cheatcode.py
+++ b/protostar/starknet/cheatcode.py
@@ -42,7 +42,9 @@ class Cheatcode(BusinessLogicSysCallHandler, HintLocal):
         self.state = syscall_dependencies["state"]
         self.general_config = syscall_dependencies["general_config"]
         self.execute_entry_point_cls = syscall_dependencies["execute_entry_point_cls"]
-        abi = self.state.get_abi_from_contract_address(syscall_dependencies["contract_address"])
+        abi = self.state.get_abi_from_contract_address(
+            syscall_dependencies["contract_address"]
+        )
         self._add_event_abi_to_state(abi)
 
     @abstractmethod

--- a/protostar/starknet/cheatcode.py
+++ b/protostar/starknet/cheatcode.py
@@ -7,7 +7,9 @@ from starkware.starknet.business_logic.execution.objects import (
 )
 from starkware.starknet.core.os.syscall_utils import BusinessLogicSysCallHandler
 from starkware.starknet.definitions.general_config import StarknetGeneralConfig
+from starkware.starknet.public.abi import AbiType
 from starkware.starknet.storage.starknet_storage import BusinessLogicStarknetStorage
+from starkware.starknet.testing.contract_utils import EventManager
 from typing_extensions import TypedDict
 
 from protostar.starknet.hint_local import HintLocal
@@ -40,7 +42,19 @@ class Cheatcode(BusinessLogicSysCallHandler, HintLocal):
         self.state = syscall_dependencies["state"]
         self.general_config = syscall_dependencies["general_config"]
         self.execute_entry_point_cls = syscall_dependencies["execute_entry_point_cls"]
+        abi = self.state.get_abi_from_contract_address(syscall_dependencies["contract_address"])
+        self._add_event_abi_to_state(abi)
 
     @abstractmethod
     def build(self) -> Callable[..., Any]:
         ...
+
+    def _add_event_abi_to_state(self, abi: AbiType):
+        event_manager = EventManager(abi=abi)
+        self.state.update_event_selector_to_name_map(
+            # pylint: disable=protected-access
+            event_manager._selector_to_name
+        )
+        # pylint: disable=protected-access
+        for event_name in event_manager._selector_to_name.values():
+            self.state.event_name_to_contract_abi_map[event_name] = abi

